utils/author_catalog.py

```diff
@@ -0,0 +1,66 @@
+import collections
+import csv
+from pathlib import Path
+from typing import Dict
+
+catalog_file_path = str(Path(__file__).parents[1]) + "/resources/" + "pg_catalog.csv"
+
+author_list = (
+    "Jefferson, Thomas, 1743-1826",
+    "Stevenson, Robert Louis, 1850-1894",
+)  # sirasiyla 1, 2, sonuc cikmasi gerekiyor.
+
+
+def read_csv(filepath=catalog_file_path):
+    lines = csv.reader(open(filepath), delimiter=",")
+    next(lines)  # skip the header.
+    for line in lines:
+        # data has some random new lines. Replace them with white space, then strip it.
+        yield list(map(lambda s: s.replace("\n", " ").strip(), line))
+
+
+def parse_data_row(row: list):
+    return (
```

Comment:
Create a `dataclass` here to capture all these info. Read this first: https://realpython.com/python-data-classes/. If we encapsulate these data as a class object, we can use `entry.book_number`, `entry.authors` instead of row[0], etc later. dataclasses provide easier way to deal with these kind of information.